functests/1_performance/netqueues.go

```diff
@@ -0,0 +1,350 @@
+package __performance
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/kubernetes/pkg/kubelet/cm/cpuset"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	performancev2 "github.com/openshift-kni/performance-addon-operators/api/v2"
+	testutils "github.com/openshift-kni/performance-addon-operators/functests/utils"
+	testclient "github.com/openshift-kni/performance-addon-operators/functests/utils/client"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/cluster"
+	testlog "github.com/openshift-kni/performance-addon-operators/functests/utils/log"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/nodes"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/pods"
+	"github.com/openshift-kni/performance-addon-operators/functests/utils/profiles"
+)
+
+var _ = Describe("[ref_id: 40307][pao]Resizing Network Queues", func() {
+	var workerRTNodes []corev1.Node
+	var err error
+	var net bool = true
+	var profile, initialProfile *performancev2.PerformanceProfile
+	workerRTNodes, err = nodes.GetByLabels(testutils.NodeSelectorLabels)
+	Expect(err).ToNot(HaveOccurred())
+	workerRTNodes, err = nodes.MatchingOptionalSelector(workerRTNodes)
+	Expect(err).ToNot(HaveOccurred())
+	nodeLabel := testutils.NodeSelectorLabels
+	profile, err = profiles.GetByNodeLabels(nodeLabel)
+	Expect(err).ToNot(HaveOccurred())
+	testutils.BeforeAll(func() {
+		isSNO, err := cluster.IsSingleNode()
+		Expect(err).ToNot(HaveOccurred())
+		RunningOnSingleNode = isSNO
+		By("Backing up the profile")
+		initialProfile = profile.DeepCopy()
+	})
+
+	BeforeEach(func() {
+		profile, err = profiles.GetByNodeLabels(nodeLabel)
+		Expect(err).ToNot(HaveOccurred())
+		if profile.Spec.Net == nil {
+			By("Enable UserLevelNetworking in Profile")
+			profile.Spec.Net = &performancev2.Net{
+				UserLevelNetworking: &net,
+			}
+			By("Updating the performance profile")
+			profiles.UpdateWithRetry(profile)
+		}
+	})
+
+	AfterEach(func() {
+		By("Reverting the Profile")
+		spec, err := json.Marshal(initialProfile.Spec)
+		Expect(err).ToNot(HaveOccurred())
+		Expect(testclient.Client.Patch(context.TODO(), profile,
```

Comment:
is this still needed ?  you are updating the profile before each,  if its still needed i wonder if we need to get the profile first before patching it  (e.g. do profile, err = profiles.GetByNodeLabels(nodeLabel) before patching)

Reply:
Yes, i am updating the profile in BeforeEach if the spec.userLevelNetworking is nil . I will update the code to get Profile

Reply:
@yanirq Actually i just checked by printing the profile spec , we don't need to  add the profile, err = profiles.GetByNodeLabels(nodeLabel) in the Aftereach section as the profile object has the latest changes from the last run test. Actually adding the line profile, err = profiles.GetByNodeLabels(nodeLabel) is throwing the error:
```
    Timed out after 60.000s.
    Expected
        <*errors.StatusError | 0xc0003a3900>: {
            ErrStatus: {
                TypeMeta: {Kind: "", APIVersion: ""},
                ListMeta: {
                    SelfLink: "",
                    ResourceVersion: "",
                    Continue: "",
                    RemainingItemCount: nil,
                },
                Status: "Failure",
                Message: "Operation cannot be fulfilled on performanceprofiles.performance.openshift.io \"performance\": the object has been modified; please apply your changes to the latest version and try again",
                Reason: "Conflict",
                Details: {
                    Name: "performance",
                    Group: "performance.openshift.io",
                    Kind: "performanceprofiles",
                    UID: "",
                    Causes: nil,
                    RetryAfterSeconds: 0,
                },
                Code: 409,
            },
        }
    to be nil
```

Reply:
ack. lets keep this then and revisit if we get flaky results or unexpected outcomes in BM